server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1911,6 +1930,60 @@ public static Metadata fromXContent(XContentParser parser) throws IOException {
             XContentParserUtils.ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.nextToken(), parser);
             return builder.build();
         }
+
+        private IndexMetadata reuseMappings(IndexMetadata indexMetadata) {
```

Comment:
One more thing, `cache` (or whatever we might rename it to) should have a little bit of JavaDoc that roughly explains what's going on here.

Reply:
This was done as part of the rename commit: 3bd8a655ddc506f88297f79d9339138cd198f73b